Diagnostic.py

```diff
@@ -8,6 +8,9 @@
 from scipy.sparse.linalg import eigs,eigsh
 import plotly.graph_objects as go
 
+'''This module purpose is to analyse neural network properties and interaction between them in case'''
+
+'''Gradient information'''
 
 def grad_info(model,t):
   total_grad = []

```

Comment:
Typically python docstrings are used like this (feel free to look at other people's code to see what the common best practices are)
```
def grad_info(model, t):
    """Gradient information"""
    total_grad = []
```

Instead of "Gradient information", you could write some short sentence what this function does: e.g. `Create gradient information`